src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/GalleryTemplatesClient.cs

```diff
@@ -401,6 +404,15 @@ private Type GetParameterType(string resourceParameterType)
             {
                 typeObject = typeof(bool);
             }
+            else if (resourceParameterType.Equals(objectType, StringComparison.OrdinalIgnoreCase)
+                || resourceParameterType.Equals(secureObjectType, StringComparison.OrdinalIgnoreCase))
+            {
+                typeObject = typeof(Hashtable);
+            }
+            else if (resourceParameterType.Equals(arrayType, StringComparison.OrdinalIgnoreCase))
+            {
+                typeObject = typeof(object[]);
```

Comment:
For Json "array" type, we can't map it to Hashtable[]. Here we are actually binding to a dynamic parameter in powershell. So it's for something like "@('abc', 'def')", which is maps to object[].